livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py

```diff
@@ -524,8 +616,11 @@ def __init__(
         self._tool_call_id: str | None = None
         self._fnc_name: str | None = None
         self._fnc_raw_arguments: str | None = None
+        self._tool_index: int | None = None
 
     async def _main_task(self) -> None:
+        if hasattr(self._llm._client, "_refresh_credentials"):
```

Comment:
Ideally we would have used a subclass to add this new function. But it is fine to merge so feel free to ignore